Update multisearcher documentation (#4320)
umbraco/UmbracoDocs: Reference/Searching/Examine/pdfindex-multisearcher.md

﻿---
versionFrom: 9.0.0
versionTo: 10.0.0
---

# PDF indexes and multisearchers

If you want to index PDF files and search for them you will need to use the [UmbracoExamine.Pdf extension package](https://github.com/umbraco/UmbracoExamine.PDF).

## Installation

Install with Nuget:
`dotnet add package Umbraco.ExaminePDF`

> You will then have a new Examine index called "PDFIndex" available.

Reviewer: ```suggestion
This will create a new Examine index called "PDFIndex", which will appear in "Examine Management" under the "Settings" section. Using this index you can start searching the contents of any PDF files uploaded to the media section.

![image](https://user-images.githubusercontent.com/7405322/189886089-d23b45c7-814b-4101-b143-31c5cd9fa655.png)

```

Reviewer: Hm the screenshot doesn't seem to play nice in the comment above. You will have to open it explicitly to see it I suppose 😞

## Multi index searchers

> 

Reviewer: Are multi index searchers explained elsewhere? Otherwise it might be good with a little intro to what they are before we dive right in.

You can register a multi index searcher with the ExamineManager on startup like:

```csharp
using Examine;
using Umbraco.Cms.Core;
using Umbraco.Cms.Core.Composing;
using Umbraco.Cms.Core.DependencyInjection;
using UmbracoExamine.PDF;

namespace MySite.MyCustomIndex
{
    [ComposeAfter(typeof(ExaminePdfComposer))]
    public class ExamineComposer : IComposer
    {
        public void Compose(IUmbracoBuilder builder)
        {
            builder.Services.AddExamineLuceneMultiSearcher("MultiSearcher", new[] {Constants.UmbracoIndexes.ExternalIndexName, PdfIndexConstants.PdfIndexName});
        }
    }

> }}

Reviewer: ```suggestion
}
```

Reviewer: One bracket too many 😄

```

> With this approach, the multisearcher will show up in the Examine dashboard and it can be resolved from the ExamineManager like:

Reviewer: ```suggestion
With this approach, the multi index searcher will show up in the "Examine Management" dashboard.

![image](https://user-images.githubusercontent.com/7405322/189887744-af2d8e69-4807-4407-868d-b43e9fa9518d.png)

The multi index searcher can be resolved in code from the ExamineManager like this:
```

Reviewer: Again with the screenshot 🤦‍♂️

﻿---
versionFrom: 9.0.0
versionTo: 10.0.0
---

# PDF indexes and multisearchers

If you want to index PDF files and search for them you will need to use the [UmbracoExamine.Pdf extension package](https://github.com/umbraco/UmbracoExamine.PDF).

## Installation

> Install with Nuget:

Reviewer 2: ```suggestion
Install with NuGet:
```

`dotnet add package Umbraco.ExaminePDF`

This will create a new Examine index called "PDFIndex", which will appear in "Examine Management" dashboard under the "Settings" section. Using this index you can start searching the contents of any PDF files uploaded to the media section.

![image](https://user-images.githubusercontent.com/7405322/189886089-d23b45c7-814b-4101-b143-31c5cd9fa655.png)

> ## Multi index searchers

Reviewer 2: ```suggestion
## Multi-index searchers
```

> A multi index searcher, is a searcher, than can search multiple indexes, this can be really helpful when you for example want to search both the external and internal indexes.

Reviewer 2: ```suggestion
A multi-index searcher is a searcher that can search multiple indexes. This can be helpful when you for example want to search both the external and internal indexes.
```

> You can register a multi index searcher with the ExamineManager on startup like:

Reviewer 2: ```suggestion
You can register a multi-index searcher with the ExamineManager on startup like:
```

```csharp
using Examine;
using Umbraco.Cms.Core;
using Umbraco.Cms.Core.Composing;
using Umbraco.Cms.Core.DependencyInjection;
using UmbracoExamine.PDF;

namespace MySite.MyCustomIndex
{
    [ComposeAfter(typeof(ExaminePdfComposer))]
    public class ExamineComposer : IComposer
    {
        public void Compose(IUmbracoBuilder builder)
        {
            builder.Services.AddExamineLuceneMultiSearcher("MultiSearcher", new[] {Constants.UmbracoIndexes.ExternalIndexName, PdfIndexConstants.PdfIndexName});
        }
    }
}
```

> With this approach, the multi index searcher will show up in the "Examine Management" dashboard.

Reviewer 2: ```suggestion
With this approach, the multi-index searcher will show up in the "Examine Management" dashboard.
```

![image](https://user-images.githubusercontent.com/7405322/189887744-af2d8e69-4807-4407-868d-b43e9fa9518d.png)

> The multi index searcher can be resolved in code from the ExamineManager like this:

Reviewer 2: ```suggestion
The multi-index searcher can be resolved in code from the ExamineManager like this:
```
